src/h5web/vis-packs/core/rgb/utils.ts

```diff
@@ -0,0 +1,14 @@
+import ndarray from 'ndarray';
+import { assign } from 'ndarray-ops';
+
+export function flipLastDimension(value: number[], dims: number[]) {
+  const baseArray = ndarray(value, dims);
+  const steps = dims.map(() => 1);
+  steps[steps.length - 1] = -1;
+
+  const flippedView = baseArray.step(...steps);
+  const flippedArray = ndarray([], dims);
+  assign(flippedArray, flippedView);
+
+  return flippedArray.data as number[];
+}
```

Comment:
Based on the same principles as `applyMapping` https://github.com/silx-kit/h5web/blob/main/src/h5web/vis-packs/core/utils.ts#L321

The last dimension is stepped in reversed order (step = -1) to generate a **view** (i.e. `baseArray.data` is unchanged) with the last dimension flipped. The extraction of `data` is done by assigning in a new `ndarray`

Reply:
Makes sense. Might be worth making a utility for this, which would take a `view` and `dims` and return a new, assigned ndarray.